fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java

```diff
@@ -124,14 +124,33 @@ public void setMaintainHistory(boolean maintainHistory) {
      * @return a future that completes with the Protobuf form of the meta-data
      */
     public CompletableFuture<RecordMetaDataProto.MetaData> loadAndSetCurrent(boolean checkCache, int currentVersion) {
+        return loadAndSetCurrent(checkCache, currentVersion, null);
+    }
+
+    /**
+     * Load current meta-data from store and set for <code>getRecordMetaData</code>.
+     *
+     * <p>
+     * This can optionally take a local file descriptor. A record store created off of the meta-data store may not be
+     * able to store a record created by a statically-generated proto file because the meta-data and record have
+     * mismatched descriptors. Using this method, the meta-data can use the same version of the descriptor as the record.
+     * See {@link RecordMetaDataBuilder#setLocalFileDescriptor(Descriptors.FileDescriptor)} for more information.
+     * </p>
+     *
+     * @param checkCache {@code true} if the cache should be checked first
+     * @param currentVersion the version to load
+     * @param localFileDescriptor local file descriptor of the record meta-data.
+     * @return a future that completes with the Protobuf form of the meta-data
+     */
+    public CompletableFuture<RecordMetaDataProto.MetaData> loadAndSetCurrent(boolean checkCache, int currentVersion, @Nullable Descriptors.FileDescriptor localFileDescriptor) {
```

Comment:
This doesn't quite feel right. I think the local file descriptor is a more intrinsic piece of state in the meta-data store, like the list of dependency file descriptors. So I'd expect it to have a similar lifetime.